### PR TITLE
Run a subset of tests across Linux, MacOS & Windows

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
- Refactor test workflows to enable running unit(ish) tests on Linux, MacOS & Windows
- Properly regenerate certs & keys and add script for doing it.
- Fix or exclude a handful of tests when running on MacOS as it's more strict around cert trust.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: api_generator/Cargo.toml [hunk 1]
```diff
@@ -9,9 +9,9 @@ edition = "2018"
 license = "Apache-2.0"
 
 [dependencies]
+anyhow = "1.0"
```
Comment (reta) [inline api_generator/Cargo.toml:12]: This change seems to be not related (reasonable simplification of error handling), could we move it off to separate pull request? ty

Comment (Xtansia) [reply api_generator/Cargo.toml]: Fair enough, have split those changes out

File: .ci/certs/generate-certs.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
```
Comment (reta) [inline .ci/certs/generate-certs.sh:1]: These scripts are not used, right? (I mean this is one time thing and may be needed in the future but not part of the CI)

File: .ci/run-opensearch.sh [hunk 3]
```diff
@@ -56,7 +56,7 @@ for (( i=0; i<$NUMBER_OF_NODES; i++, http_port++ )); do
   echo -e "\033[34;1mINFO:\033[0m Starting container $node_name \033[0m"
   set -x
   if [[ "$SECURE_INTEGRATION" == "true" ]]; then
-    healthcmd="curl -vvv -s --insecure -u admin:admin --fail https://localhost:9200/_cluster/health || exit 1"
+    healthcmd="curl -vvv -s --cacert /usr/share/opensearch/config/root-ca.pem -u admin:admin --fail https://localhost:9200/_cluster/health || exit 1"
```
Comment (reta) [inline .ci/run-opensearch.sh:59]: :+1:

File: .ci/certs/generate-certs.sh (hunk 2 above)
Comment (Xtansia) [reply .ci/certs/generate-certs.sh:1]: They're not actively run no, they're more for posterity and so that if at some point we need to modify / regenerate them it's easier. ie. future me doesn't have to rediscover how they link together or what the right commands & config is 🥲

**Comment by reta** [comment]
@Xtansia for general knowledge 

> Fix or exclude a handful of tests when running on MacOS as it's more strict around cert trust.

Could you please share the issue here? (my apologies, no MacOS box at the moment available).

File: .ci/certs/generate-certs.sh (hunk 2 above)
Comment (reta) [reply .ci/certs/generate-certs.sh:1]: Totally makes sense, thank you.
